### PR TITLE
Add framework target for netcoreapp3.1 and associated appveyor changes

### PR DESCRIPTION
**Pull request by seif**
With additional compilation and references

### CODE REVIEW

**Comment by seif** [comment]
@davidrevoledo not sure what is going on with appveyor, is it using the files in the root? Can you have a look?

**Comment by davidrevoledo** [comment]
@seif will take a look thanks for the PR

**Comment by seif** [comment]
@davidrevoledo anything I can do to help? If you let me know how you want the build pipeline to work I can look at setting up appveyor.

**Comment by davidrevoledo** [comment]
@seif  I can add you to manage the auto build 
the error right now is
```
C:\Program Files\dotnet\sdk\2.2.402\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.TargetFrameworkInference.targets(137,5): error NETSDK1045: The current .NET SDK does not support targeting .NET Core 3.1.  Either target .NET Core 2.2 or lower, or use a version of the .NET SDK that supports .NET Core 3.1. [C:\projects\aspnetcore-autohealthcheck\src\AspNetCore.AutoHealthCheck.Abstractions\AspNetCore.AutoHealthCheck.Abstractions.csproj]
Build FAILED.
C:\Program Files\dotnet\sdk\2.2.402\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.TargetFrameworkInference.targets(137,5): error NETSDK1045: The current .NET SDK does not support targeting .NET Core 3.1.  Either target .NET Core 2.2 or lower, or use a version of the .NET SDK that supports .NET Core 3.1. [C:\projects\aspnetcore-autohealthcheck\src\AspNetCore.AutoHealthCheck.Abstractions\AspNetCore.AutoHealthCheck.Abstractions.csproj]
    0 Warning(s)
    1 Error(s)
Time Elapsed 00:00:01.17

```

**Comment by seif** [comment]
Thanks, yeah that would help. As I need to change appveyor build image and modifying the files isn't working

**Comment by seif** [comment]
@davidrevoledo I've setup an appveyor build to fix things. I am not sure what your original intention was or how packages are currently published but this pr configures the build to:
 run for master/dev branches (so we have 1 file instead of the 2)
run in ubuntu and windows 2019 (2019 appveyor image is required for netcoreapp3.1)
pack nuget packages and publish them (this can run on ubuntu or linux, whatever)
Havent actually tested the pulblishing to nuget but put instructions on how to populate the secret key

Thanks

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+for:
+-
+  matrix:
+    only:
+      - image: Ubuntu
+  deploy:
+  - provider: NuGet
+    name: production
+    api_key:
+      secure: "Use 'Encrypt configuration data' page in AppVeyor (Account → Encrypt YAML) to encrypt your nuget API key and place it here"
```
Comment (seif) [inline appveyor.yml:41]: @davidrevoledo doesn't look like nuget publishing was previously configured. I could remove this to simplify this pr.

**Comment by seif** [comment]
@davidrevoledo thanks!

Can you add me as owner to the nuget package https://www.nuget.org/packages/AspNetCore.AutoHealthCheck/ so I can setup an api key that key that will publish from appveyor? Looks like giving me access to appveyor to encrypt the key is a fiddly process. So I've removed the nuget publishing from the change in this PR, will open another PR once the api key is sorted.

**Comment by seif** [comment]
ok to merge/ship this meanwhile? :)

**Comment by seif** [comment]
@davidrevoledo you happy for me to merge this?

**Comment by davidrevoledo** [comment]
sure thing @seif

**Comment by seif** [comment]
@davidrevoledo thanks, I've merged also merged #13 however can't publish the nuget package as I am not an owner in nuget. Can you add my user 'seifattar' as owner to https://www.nuget.org/packages/AspNetCore.AutoHealthCheck/Manage I can then setup a token in #14 and release

**Pull request merged**